source/simulator/core.cc

```diff
@@ -2056,26 +2102,61 @@ namespace aspect
         initialize_tracers ();
 
         computing_timer.exit_section();
+
+        if (parameters.nonlinear_solver == NonlinearSolver::Binary_input)
+        {
+          const std::string filename = parameters.binary_data_directory + "/fields-" +
+                                       Utilities::int_to_string(parameters.binary_data_ts_number, 5) + ".bin";
+          std::ifstream input_stream (filename);
+          AssertThrow(input_stream,
+                      ExcMessage(std::string("Failed to read the file, " + filename +
+                                             " for the time step number, " + Utilities::int_to_string(parameters.binary_data_ts_number) + " using Nonlinear solver mode \"Binary_input\".")));
+          Utilities::BinaryInputFields input_fields;
+          boost::archive::binary_iarchive ia(input_stream);
+          ia >> input_fields;
+          time = input_fields.time;
+          time_step = input_fields.time_step;
+          timestep_number = input_fields.timestep_number;
+          old_time_step = input_fields.old_time_step;
+          input_stream.close();
+        }
       }
 
     // start the principal loop over time steps:
-    do
-      {
-        start_timestep ();
+    do {
+      start_timestep();
 
-        // then do the core work: assemble systems and solve
-        solve_timestep ();
+      // then do the core work: assemble systems and solve
+      solve_timestep();
 
-        pcout << std::endl;
+      pcout << std::endl;
 
-        // update the time step size
-        // for now the bool (convection/conduction dominated)
-        // returned by compute_time_step is unused, will be
-        // added to statistics later
+      // update the time step size
+      // for now the bool (convection/conduction dominated)
+      // returned by compute_time_step is unused, will be
+      // added to statistics later
+      if (parameters.nonlinear_solver != NonlinearSolver::Binary_input) {
         old_time_step = time_step;
-        time_step = std::min (compute_time_step().first,
-                              parameters.maximum_time_step);
+        time_step = std::min(compute_time_step().first,
+                             parameters.maximum_time_step);
         time_step = termination_manager.check_for_last_time_step(time_step);
+      }
+      else {
+        const std::string filename = parameters.binary_data_directory + "/fields-" +
```

Comment:
When me and Gerry were initially using the ascii data model of prescribed stokes solution, we would set the global start time parameter and read the appropriate time step data, i.e starting from the 50th time step number. This run also contained particles that were being advected by the prescribed stokes solution which would be updated at each time step. After 200 or so time steps, we visually noticed that the particles no longer followed the flow. The reason was a direct result of the time step values not being accurate enough. We later sent in a pull request that increased the accuracy of the time step values in the log output. When we set the start time to be these time values up to 14 digits of accuracy, down a 1000+ time step numbers, the computed time began to differ compared to the first computational pass. I can run a simple test case to check that we do not need to set the time step explicitly however.